Source/JavaScriptCore/wasm/WasmAirIRGenerator.cpp

```diff
@@ -2187,6 +2158,54 @@ void AirIRGenerator::sanitizeAtomicResult(ExtAtomicOpType op, Type valueType, Tm
     }
 }
 
+void AirIRGenerator::emitWriteBarrier(TypedTmp cell, ShouldUseCellAsInstance cellIsInstance)
+{
+    auto instanceCell = g64();
+    auto vm = g64();
+    auto cellState = g32();
+    auto threshold = g32();
+
+    BasicBlock* fenceCheckPath = m_code.addBlock();
+    BasicBlock* fencePath = m_code.addBlock();
+    BasicBlock* doSlowPath = m_code.addBlock();
+    BasicBlock* continuation = m_code.addBlock();
+
+    if (cellIsInstance == ShouldUseCellAsInstance::Yes)
```

Comment:
I am not a big fan of this flag, it is kind of confusing. Why take in an instance tmp, and make a helper function for the cellIsInstance::no case that loads it? emitWriteBarrierForJSWrapper is the only user of that case anyway, right?

Reply:
Yeah, I see, I removed the flag, so now it should be less confusing: `emitBarrier` now takes two arguments `emitBarrier(cell, instance)`, in case of *jsWrapper* `cell` and `instance` are equal, but for other, more grained applications they can differ.